ldes-server-infra-postgres/postgres-pagination-repository/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/pagination/postgres/batch/PageRelationProcessorTest.java

```diff
@@ -30,82 +27,87 @@ class PageRelationProcessorTest {
 	private static final long CHILD_PAGE_ID = PARENT_PAGE_ID + 1;
 
 	@Mock
-	private JdbcTemplate jdbcTemplate;
+	private PageRepository pageRepository;
+	@Mock
+	private PageRelationRepository pageRelationRepository;
 	@InjectMocks
 	private PageRelationProcessor pageRelationProcessor;
 
 	@Test
 	void given_FullPage_when_Process_then_CreateNewPage() {
-		final PageAssignment expectedPageAssignment = new PageAssignment(CHILD_PAGE_ID, BUCKET_ID, UNPROCESSED_MEMBER_COUNT);
 		final Page parentPage = new Page(PARENT_PAGE_ID, BUCKET_ID, "/mobility-hindrances/by-page?pageNumber=2", PAGE_SIZE, PAGE_SIZE);
-		when(jdbcTemplate.queryForObject(SELECT_UNPROCESSED_MEMBER_COUNT, Integer.class, BUCKET_ID)).thenReturn(UNPROCESSED_MEMBER_COUNT);
-		when(jdbcTemplate.update(any(PreparedStatementCreator.class), any(GeneratedKeyHolder.class)))
-				.thenAnswer(invocationOnMock -> {
-					invocationOnMock.getArgument(1, GeneratedKeyHolder.class).getKeyList().add(Map.of("page_id", CHILD_PAGE_ID));
-					return 1;
-				});
-
-		final List<PageAssignment> result = pageRelationProcessor.process(parentPage);
-
-		verify(jdbcTemplate).update(INSERT_PAGE_RELATION_SQL, PARENT_PAGE_ID, CHILD_PAGE_ID);
-		assertThat(result).containsExactly(expectedPageAssignment);
+		when(pageRepository.getOpenPage(BUCKET_ID)).thenReturn(parentPage);
+		when(pageRepository.createPage(BUCKET_ID, parentPage.createChildPartialUrl().asString())).thenReturn((int) CHILD_PAGE_ID);
+
+		final List<PageAssignment> result = pageRelationProcessor.process(
+				Collections.nCopies(UNPROCESSED_MEMBER_COUNT, new UnpagedMember(0, BUCKET_ID)));
+
+		verify(pageRepository).createPage(parentPage.getBucketId(), parentPage.createChildPartialUrl().asString());
+		verify(pageRelationRepository).insertGenericBucketRelation(eq(parentPage.getId()), anyLong());
+		assertThat(result).isNotNull();
+		assertThat(result)
+				.filteredOn("pageId", CHILD_PAGE_ID)
+				.filteredOn("bucketId", BUCKET_ID)
```

Comment:
Could a Function<PageAssignment, ? extends Object> be used (e.g. PageAssigment::pageId) be used instead of the propertyOrFieldName?